lib/rtsp.c

```diff
@@ -822,7 +863,59 @@ CURLcode Curl_rtsp_parseheader(struct Curl_easy *data, char *header)
       (data->set.str[STRING_RTSP_SESSION_ID])[idlen] = '\0';
     }
   }
+  else if(checkprefix("Transport:", header)) {
+    CURLcode result;
+    result = rtsp_parse_transport(data, header + 10);
+    if(result)
+      return result;
+  }
+  return CURLE_OK;
+}
+
+static
+CURLcode rtsp_parse_transport(struct Curl_easy *data, char *transport)
+{
+  /* e.g.: ' RTP/AVP/TCP;unicast;interleaved=5-6' */
+  char *start;
+  char *end;
+  start = transport;
+  while(start && *start) {
+    while(*start && ISBLANK(*start) )
+      start++;
+    end = strchr(start, ';');
+    if(checkprefix("interleaved=", start)) {
```

Comment:
A slightly odd thing about if there are more than one `interleaved=` or more than one Transport header is that you then we basically do an OR of the channel ranges, which I cannot see that the RFC says anything about.

Reply:
The Transport header is restricted to describing a single media stream.  RTSP can also control multiple streams as a single entity (Reference RFC7826 - 18.54).
During an RTSP session, the multiple SETUP requests containing the Transport header can each control the multiple streams separately. The channel number range selected by the interleaved parameter in the Transport header of each stream should not overlap, otherwise the client cannot decide which stream to associate with the interleaved data received. All of these selected channel numbers are valid, so we  do an OR of the channel ranges.

More complexity,  The Transport header MAY also be used in subsequent SETUP requests to change transport parameters. (Reference RFC7826 - 18.54)
If we want to accurately record all the final selected channel numbers, and can exclude those that have also been changed, we need to save the channel number selected by each stream separately. Which will have to add dynamic memory allocation (because the number of maximum possible streams is unknown) and greatly increase the complexity. On the other hand, the behavior of modifying the transport parameters may be very rare, and we only use the channel number for data validity detection, I think that it is not necessary to record the accurate final channel number.